### ISSUE TITLE
Menu - fullscreen story- focus doesn't land correctly with keyboard navigation

### ISSUE THREAD

**Issue by dawidzarzycki** [Sage/carbon#4287]
### Current behaviour

If we open storybook and we go to Menu - fullscreen story and we tab to the first Menu button and press enter and then we will press tab one time, we won't land on any item inside of MenuFullScreen. We will go to the next menu button that is under the fullscreen Menu. It looks like the focus won't target on fullscreen Menu first and we have to tab more times to land into the FullScreenMenu content

### Expected behaviour

If we open full screen with Menu button with keyboard navigation (enter) we should have active MenuFullScreen window already. So we can tab through the FullScreenMenu content instead of content under the menu

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/design-system-menu--fullscreen-view

### Carbon Version

80.1.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by dawidzarzycki** [comment]
FE-4252

**Comment by edleeks87** [comment]
@harpalsingh @tempertemper would you guys mind giving a steer on this please, more info is on the jira ticket

**Comment by sjohnson-sage** [comment]
@harpalsingh @tempertemper would you both mind having a look at this when you get a chance please? Thanks

**Comment by tempertemper** [comment]
This component is effectively a modal, so the content behind the menu should not be able to be interacted with while it's open, either by keyboard or pointer. Keyboard focus should be 'trapped' in the menu/modal until it is closed with the X button, with the <kbd>esc</kbd> key, or a navigation link is followed and the user goes to a new page/view.
